files/src/main/java/com/fincity/saas/files/service/FilesAccessPathService.java

```diff
@@ -303,7 +303,8 @@ public Mono<Integer> delete(ULong id) {
 	public Mono<Boolean> hasReadAccess(String actualPath, String clientCode, FilesAccessPathResourceType resourceType) {
 
 		String path = actualPath.endsWith("/") ? actualPath.substring(0, actualPath.length() - 1) : actualPath;
-		Mono<String> mKey = cacheService.makeKey(clientCode, ":", resourceType, ":read:", path);
+
+		Mono<String> mKey = Mono.just(clientCode + ":" + resourceType + ":read:" + path);
 
 		return mKey.flatMap(e -> cacheService.<Boolean>get(CACHE_NAME_ACCESS_PATH, e))
```

Comment:
We need to update the way we are storing the values in the Cache. Use cacheOrGet method in the cache services.